languages/javascript/exercises/concept/errors/.docs/instructions.md

```diff
@@ -0,0 +1,42 @@
+Elena is the new quality manager of a newspaper factory. As she arrived in the company, she decided to review some of the processes in the factory to see what could be improved. She found out that some technicians are checking the temperatures of the machines every 30 minutes, to make sure the machine is not overheating. She sees there a good oportunity for automation and ask you, a freelance developer, to develop a piece of software to monitor the machines' temperature.
+
+## Detect overheating
+
+After chatting with a technician, Greg, you are told that if the temperature of a machine exceeds 500°C, they should start worrying about overheating.
+
+Your task is to implement a function `reportOverheating` that takes a temperature as argument (from the machine's sensor). If the temperature exceeds the limit, you should throw an error.
+
+## Input validation
+
+After testing your code a bit, you notice that sometimes, the sensor send you a `null` value instead of the temperature. You ask Greg about it. He tells you that some machines are quite old and the sensor might not work from time to time.
+
+If that happens, you will get a `null` value and you should throw an error.
+
+## Making the errors actionnable
+
+You now have errors when the machine temperature is too high or when the temperature is `null`. That said, you would like to react to this errors in different ways, depending on the kind of error.
+
+You could rely on the error messages, but this would make your code weak as it would break if the message got updated.
+
+So to be able to do so properly, you'll improve your previous code to throw an `ArgumentError` if the temperature you receive is `null` and an `OverheatingError` if the temperature is too high.
```

Comment:
I'm wondering if the improving part is a good idea. It sounds weird to write two versions of the same function in a single file. I'm wondering how we could make this more smooth

Reply:
@SleeplessByte Have you had the time to review my new draft version ?

Reply:
@TomPradat instead of improving, you could have them write _two_ functions. One that is non-improved, and one that is improved (but for a different sensor or whatever).

Reply:
Alright, I did two different functions, one that checks the humidity percentage of a room, with a simple "Error" that should be thrown. Then the `checkTemperature` where you have to throw instances of custom Errors